py/desispec/scripts/tsnr_afterburner.py

```diff
@@ -0,0 +1,197 @@
+            tileid = cframe[0].header['TILEID']
+        
+            iin = x.replace('cframe', 'frame')
+            sky = x.replace('cframe', 'sky')
+            psf = sky.replace('sky', 'psf')
+            nea = '/project/projectdirs/desi/users/mjwilson/master_nea/masternea_{}.fits'.format(camera)  
+            ens = '/project/projectdirs/desi/users/mjwilson/tsnr-ensemble/'
+            out = args.outdir + '/tsnr/{}/{:08d}/tsnr-{}-{:08d}.fits'.format(night, expid, camera, expid)
+
+            if (not args.summary_only) & os.path.exists(out):
+                continue
+            
+            Path(os.path.dirname(out)).mkdir(parents=True, exist_ok=True)
+                
+            frame=read_frame(iin)
+            fiberflat=read_fiberflat(flat)
+            fluxcalib=read_flux_calibration(calib)
+            skymodel=read_sky(sky)
+            
+            results = calc_tsnr(frame, fiberflat=fiberflat, skymodel=skymodel, fluxcalib=fluxcalib)
+
+            if not args.summary_only:
+                # Write individual.
+                table=Table()
+                
+                for k in results:
+                    if k != 'ALPHA':
+                        table[k] = results[k].astype(np.float32)
+
+                table['FIBER']       = frame.fibermap['FIBER']
+                table['TARGETID']    = frame.fibermap['TARGETID']
+                
+                table.meta['NIGHT']  = night
+                table.meta['EXPID']  = '{:08d}'.format(expid)
+                table.meta['ALPHA']  = results['ALPHA']
+                table.meta['TILEID'] = tileid
+            
+                table.write(out, format='fits', overwrite=True)
+
+                log.info('Successfully written {}.'.format(out))
+                
+            # Append to summary. 
+            entry = Table(data=np.array(['{:08d}'.format(expid)]), names=['EXPID'])
+            
+            entry['NIGHT']  = night
+            entry['CAMERA'] = camera
+
+            keys = list(results.keys())
+            
+            for k in keys:
+                if 'TSNR' in k:
+                    sk = k.split('_')[0]    
+                    results[sk] = results[k]
+                    del results[k]
+                                
+            for k in results:                                                                                                                                                                                                
+                entry[k] = np.median(results[k].astype(np.float32))
+                
+            if summary is None:
+                summary = entry
+
+            else:
+                summary = vstack((summary, entry))
+                
+            log.info('{:08d}  {}: Reduced {} of {}.'.format(expid, cam, kk, len(sci_frames[cam])))
+
+        hdul.close()
+        
+    # 
+    summary.write('/global/cscratch1/sd/mjwilson/trash/afterburner/tsnr/summary_{}.fits'.format(cam), format='fits', overwrite=True)
```

Comment:
This hardcoded path, however, appears to be used.  Please scrub all hardcoded paths from this code and use options like `--outdir` instead.